pkg/scheduler/framework/plugins/README.md

```diff
@@ -0,0 +1,116 @@
+# Scheduler Framework Plugins
+
+## Creating a new in-tree plugin
+
+TODO: https://github.com/kubernetes/kubernetes/issues/95156
+
+## Making an in-tree plugin configurable through `KubeSchedulerConfiguration`
+
+You can give users the ability to configure parameters in scheduler plugins using
+[`KubeSchedulerConfiguration`](https://kubernetes.io/docs/reference/scheduling/config/).
+This section will cover how you can add arguments to existing in-tree plugins.
+Let's assume the plugin is called `FooPlugin` and we want to add an optional
+integer parameter named `BarParam`.
+
+First, we need to define a struct type named `FooPluginArgs`. In
+`pkg/scheduler/apis/config/types_pluginargs.go`, which defines the scheduler's
+internal struct type for passing config arguments:
+
+```go
+// Defined in pkg/scheduler/apis/config/types_pluginargs.go.
+type FooPluginArgs struct {
+	metav1.TypeMeta
+	BarParam int32
+}
+```
+
+Similarly, define `FooPluginArgs` in `k8s.io/kube-scheduler/config/v1beta1/types_pluginargs.go`.
+This time, however, to allow implicit default values for arguments, the type of
+the struct's fields may be pointers; the zero value of a pointer field (nil)
+will make the framework to use the default value. Since `BarParam` is of type
+`int32` and since we want a non-zero default value:
+
+```go
+// Defined in staging/src/k8s.io/kube-scheduler/config/v1beta1/types_pluginargs.go.
+type FooPluginArgs struct {
```

Comment:
This also needs deep copy generation comments.
```
// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
```

Reply:
Could we also add another paragraph explaining why this comment is needed. For instance:

> We add `// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object` comment to auto-generate `DeepCopy` function. More info: https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/code-generator